src/main/kotlin/no/nav/familie/ef/sak/vilkår/VurderingService.kt

```diff
@@ -166,6 +170,102 @@ class VurderingService(
+        val sivilstandErLik =
+            hentGrunnlagsdataOgValiderSivilstand(tidligereBehandlingId, nåværendeBehandlingId)
+        val (tidligereVurderinger, nåværendeVurderinger) = finnVurderingerSomSkalGjenbrukes(
+            tidligereBehandlingId,
+            sivilstandErLik,
+            forrigeVurderingIdTilNåværendeBarnMap,
+            nåværendeBehandlingId
+        )
+
+        val vurderingerSomSkalLagres = tidligereVurderinger.mapNotNull { tidligereVurdering ->
+            if (tidligereVurdering.type == VilkårType.ALENEOMSORG) {
+                val barnForVurdering = forrigeVurderingIdTilNåværendeBarnMap[tidligereVurdering.barnId]
+                val nåværendeVurdering =
+                    nåværendeVurderinger.firstOrNull { nåværendeVurdering -> nåværendeVurdering.type == tidligereVurdering.type  && nåværendeVurdering.barnId == barnForVurdering?.id }
+                if (nåværendeVurdering != null) {
+                    tidligereVurdering.copy(
+                        id = nåværendeVurdering.id,
+                        behandlingId = nåværendeBehandlingId,
+                        sporbar = nåværendeVurdering.sporbar,
+                        barnId = nåværendeVurdering.barnId,
+                    )
+                } else {
+                    null
+                }
+            } else {
+                val nåværendeVurdering = nåværendeVurderinger.first { b -> tidligereVurdering.type == b.type }
+                tidligereVurdering.copy(
+                    id = nåværendeVurdering.id,
+                    behandlingId = nåværendeBehandlingId,
+                    sporbar = nåværendeVurdering.sporbar,
+                )
+            }
+        }
+        vilkårsvurderingRepository.updateAll(vurderingerSomSkalLagres)
+    }
+
+    private fun finnBarnPåBeggeBehandlinger(
+        tidligereBehandlingId: UUID,
+        nåværendeBehandlingId: UUID
+    ): Map<UUID, BehandlingBarn> {
+        val barnPåForrigeBehandling =
+            barnService.finnBarnPåBehandling(tidligereBehandlingId)
+        val barnPåNåværendeBehandling =
+            barnService.finnBarnPåBehandling(nåværendeBehandlingId)
+        val barnPåForrigeBehandlingSomErINåværede = barnPåForrigeBehandling.filter { forrige ->
+            barnPåNåværendeBehandling.map { nåværende -> nåværende.personIdent }.contains(forrige.personIdent)
+        }
+        val barnPåNåværendeBehandlingSomVarIForrige = barnPåNåværendeBehandling.filter { nåværende ->
+            barnPåForrigeBehandling.map { forrige -> forrige.personIdent }.contains(nåværende.personIdent)
+        }
+        val forrigeMap = barnPåForrigeBehandlingSomErINåværede.associate { it.personIdent to it }
+        val forrigeVurderingIdTilNåværendeBarnMap =
+            barnPåNåværendeBehandlingSomVarIForrige.associate { forrigeMap[it.personIdent]!!.id to it }
+        return forrigeVurderingIdTilNåværendeBarnMap
+    }
```

Comment:
Kan denne forenklet, og blir dette det samme?
Notere at jeg har endret ordningen på argument, og navnet på de, de blir litt kortere, og tenker `behandlingId` alltid kan ses som nåværende? 

 ```suggestion
    private fun mapNåværendeBarnPåForrigeBarnId(
        behandlingId: UUID
        fraBehandlingId: UUID
    ): Map<UUID, BehandlingBarn> {
        val behandlingBarn = barnService.finnBarnPåBehandling(behandlingId).associateBy { it.personIdent }
        val barnPåForrigeBehandling = barnService.finnBarnPåBehandling(fraBehandlingId)

        return barnPåForrigeBehandling.mapNotNull { forrige ->
            val barn = behandlingBarn[forrige.personIdent] ?: return@mapNotNull null
            barn.id to barn
        }.toMap()
    }
```

Reply:
Koden blir mer konsis 👍 Men siden barnPåForrigeBehandling barnPåNåværendeBehandling sine ider er forskjellige selv om det er snakk om samme barn blir mappingen slik: `forrige.id to barn`